src/core/net/nat64_translator.cpp

```diff
@@ -133,14 +139,43 @@ Translator::Result Translator::TranslateFromIp6(Message &aMessage)
         ExitNow(res = kNotTranslated);
     }
 
-    mapping = FindOrAllocateMapping(ip6Header.GetSource());
```

Comment:
Can consider using `Ip6::Headers` here. I think it can simplify the code.

It parses IPv6 header first, then UDP/TCP/ICMP6 headers (based on the next header):

```
    /**
     * Parses the IPv6 and UDP/TCP/ICMP6 headers from a given message.
     *
     * @param[in] aMessage   The message to parse the headers from.
     *
     * @retval kErrorNone    The headers are parsed successfully.
     * @retval kErrorParse   Failed to parse the headers.
     */
    Error ParseFrom(const Message &aMessage);
```

Reply:
I changed the code according to your suggestion, indeed it simplifies it. For the 4 to 6 direction we don't have a similar class so I left the code as before.